src/JsonPointer/JsonPointer.cs

```diff
@@ -28,7 +27,6 @@ public class JsonPointer : IEquatable<JsonPointer>, IReadOnlyList<string>
 	public static readonly JsonPointer Empty = new();
 
 	private readonly string[] _decodedSegments;
-	private string? _plain;
```

Comment:
Surely it would help to cache the string rather than rebuilding it each time.  This would be populated either on `.Parse()` (the original string input) or lazily on `.ToString()` (if the pointer was created with `.Create()`).

Reply:
I brought back the caching and populate it in situations where this is possible without allocations. In all other cases, the cost is moved to `ToString()`